app/src/main/java/io/heckel/ntfy/service/SubscriberService.kt

```diff
@@ -247,11 +250,39 @@ class SubscriberService : Service() {
         }
     }
 
+    private fun onConnectionOpen(connectionId: ConnectionId, message: String?) {
+        Log.d(TAG, "Received open from connection ${connectionId.baseUrl} with message: $message")
+        // this check is sufficient for now, and the message can be upgraded to include other parameters in the future
+        if (message?.contains(ApiService.EVENT_OPEN_PARAM_NEW_TOPIC) == true) {
+            GlobalScope.launch(Dispatchers.IO) {
+                for (topic in connectionId.topicsToSubscriptionIds.keys) {
+                    if (connectionId.topicIsUnifiedPush[topic] == true) {
+                        Log.d(TAG, "Attempting to re-register ${connectionId.baseUrl}/$topic")
+                        io.heckel.ntfy.up.BroadcastReceiver.sendRegistration(baseContext, connectionId.baseUrl, topic)
+                        // TODO is that the right context - looks like it works???
```

Comment:
Should that work? Or do I need to make a whole chain of contexts being passed down and stored? Not really sure what context is used for when creating the repository downstream.